src/main/db/test-db.js

```diff
@@ -1,33 +1,56 @@
-import db from './db';
+import { db as config } from 'common/config';
+import { Model } from 'objection';
+import Knex from 'knex';
 
-let initialized = false;
-
-const init = async () => {
-	initialized = true;
-	await db.createInitialDb();
-};
-
-const ensureConnection = async () => {
-	try {
-		await db.knex.raw('select 1+1 as result');
-		return false;
-	} catch (err) {
-		await db.knex.client.initializePool(db.config);
-		return true;
+export class TestDb {
+	static async init() {
+		try {
+			this.knex = Knex(config);
+			this.oldKnex = Model.knex();
+			Model.knex(this.knex);
+			await this.aquireConnection();
+			await this.knex.migrate.latest();
+			await this.knex.seed.run();
+		} catch (error) {
+			console.log(error);
+		}
 	}
-};
-
-const reset = async () => {
-	let isNewConn = await ensureConnection();
-	if (!isNewConn) {
-		await db.reset();
+	static async aquireConnection(attempt = 0) {
+		try {
+			await this.knex.raw('select 1+1 as result');
+		} catch (error) {
+			if (attempt === 3) {
+				console.error('Could not establish connection', 3, 'times', error);
+				throw error;
+			}
+			return this.aquireConnection(attempt + 1);
+		}
 	}
-	await init();
-};
-
-export { init, reset };
+	static async reset() {
+		try {
+			await this.knex.destroy();
+			Model.knex(this.oldKnex);
```

Comment:
I don't understand what this is doing.

Reply:
@dfcowell that's a result of a very long fight with db connection errors in tests. This one is a leftover, will remove